src/encode.rs

```diff
@@ -12,92 +12,66 @@
 // If not, see <http://creativecommons.org/publicdomain/zero/1.0/>.
 //
 
-//! # Encoding/Decoding
-//!
-//! Functionality to encode or decode Simplicity programs. These programs
-//! are encoded bitwise rather than bytewise, so given a hex dump of a
-//! program it is not generally possible to read it visually the way you
-//! can with Bitcoin Script.
+//! # Encoding
 //!
+//! Functionality to encode Simplicity programs.
+//! These programs are encoded bitwise rather than bytewise,
+//! so given a hex dump of a program it is not generally possible
+//! to read it visually the way you can with Bitcoin Script.
 
 use std::{io, mem};
 
-use crate::bititer::BitIter;
-
-use crate::extension;
-use crate::extension::Jet as ExtNode;
-use crate::{Error, Term};
-
-use crate::core::term::UntypedProgram;
-use crate::merkle::cmr::Cmr;
-
-/// Trait for writing individual bits to some sink
-pub trait BitWrite {
```

Comment:
Removing `BitWrite` in favour of `BitWriter` is a decision worth discussing. My idea was to mimic the use of `BitIter`, which is also a struct. Any thoughts?

Reply:
Yeah, I think this is a good decision. The trait thing was probably a bit too clever.